cpp/src/arrow/adapters/orc/adapter_test.cc

```diff
@@ -226,19 +224,54 @@ std::shared_ptr<Table> GenerateRandomTable(const std::shared_ptr<Schema>& schema
   return Table::Make(schema, cv);
 }
 
+arrow::adapters::orc::WriteOptions GenerateRandomWriteOptions(int64_t num_cols) {
+  auto arrow_write_options = arrow::adapters::orc::WriteOptions();
+  arrow_write_options.batch_size = arrow::random_single_int<int64_t, int64_t>(4ull, 8ull);
+  arrow_write_options.file_version = arrow::adapters::orc::FileVersion(
+      0, arrow::random_single_int<int32_t, int32_t>(11, 12));
+  arrow_write_options.stripe_size =
+      arrow::random_single_int<int64_t, int64_t>(4ull, 128ull);
+  arrow_write_options.compression_block_size =
+      arrow::random_single_int<int64_t, int64_t>(4ull, 128ull);
+  arrow_write_options.row_index_stride =
+      arrow::random_single_int<int64_t, int64_t>(0, 128ull);
+  arrow_write_options.compression =
+      arrow::random_single_int<int8_t, arrow::Compression::type>(0, 2);
```

Comment:
Not all compression kinds may be enabled, so you'd need to skip the test selectively if the given compression type isn't enabled.

Reply:
Currently it is using the fact that the first 3 compression types are enabled.

Reply:
Fixed